test/suites/integration/conversion_webhook_test.go

```diff
@@ -0,0 +1,210 @@
+						ExpireAfter:         karpv1beta1.NillableDuration{Duration: lo.ToPtr(time.Hour)},
+					},
+					Template: karpv1beta1.NodeClaimTemplate{
+						Spec: karpv1beta1.NodeClaimSpec{
+							NodeClassRef: &karpv1beta1.NodeClassReference{
+								Name: "test-nodeclass",
+							},
+							Requirements: []karpv1beta1.NodeSelectorRequirementWithMinValues{
+								{
+									NodeSelectorRequirement: corev1.NodeSelectorRequirement{
+										Key:      corev1.LabelOSStable,
+										Operator: corev1.NodeSelectorOpIn,
+										Values:   []string{string(corev1.Linux)},
+									},
+								},
+							},
+							Kubelet: &karpv1beta1.KubeletConfiguration{
+								MaxPods:     lo.ToPtr(int32(110)),
+								PodsPerCore: lo.ToPtr(int32(10)),
+								SystemReserved: map[string]string{
+									string(corev1.ResourceCPU):              "200m",
+									string(corev1.ResourceMemory):           "200Mi",
+									string(corev1.ResourceEphemeralStorage): "1Gi",
+								},
+								KubeReserved: map[string]string{
+									string(corev1.ResourceCPU):              "200m",
+									string(corev1.ResourceMemory):           "200Mi",
+									string(corev1.ResourceEphemeralStorage): "1Gi",
+								},
+								EvictionHard: map[string]string{
+									"memory.available":   "5%",
+									"nodefs.available":   "5%",
+									"nodefs.inodesFree":  "5%",
+									"imagefs.available":  "5%",
+									"imagefs.inodesFree": "5%",
+									"pid.available":      "3%",
+								},
+								EvictionSoft: map[string]string{
+									"memory.available":   "10%",
+									"nodefs.available":   "10%",
+									"nodefs.inodesFree":  "10%",
+									"imagefs.available":  "10%",
+									"imagefs.inodesFree": "10%",
+									"pid.available":      "6%",
+								},
+								EvictionSoftGracePeriod: map[string]metav1.Duration{
+									"memory.available":   {Duration: time.Minute * 2},
+									"nodefs.available":   {Duration: time.Minute * 2},
+									"nodefs.inodesFree":  {Duration: time.Minute * 2},
+									"imagefs.available":  {Duration: time.Minute * 2},
+									"imagefs.inodesFree": {Duration: time.Minute * 2},
+									"pid.available":      {Duration: time.Minute * 2},
+								},
+								EvictionMaxPodGracePeriod:   lo.ToPtr(int32(120)),
+								ImageGCHighThresholdPercent: lo.ToPtr(int32(50)),
+								ImageGCLowThresholdPercent:  lo.ToPtr(int32(10)),
+								CPUCFSQuota:                 lo.ToPtr(false),
+							},
+						},
+					},
+				},
+			}
+
+			// Use a deepcopy to make sure the nodePool object is not populated with the returned object from the APIServer
+			env.ExpectCreated(storedv1beta1NodePool.DeepCopy())
+			v1beta1NodePool := env.ExpectExists(storedv1beta1NodePool.DeepCopy()).(*karpv1beta1.NodePool)
+			Expect(v1beta1NodePool.Generation).To(BeNumerically("==", 1))
+
+			// Second apply of the same NodePool does not increase the generation
+			env.ExpectUpdated(storedv1beta1NodePool.DeepCopy())
```

Comment:
This seems good. Should we also validate that the object that we pull from the apiserver on v1beta1 is the same as the one that we applied -- regardless of whether the generation increases or not.